storage/series/buffer.go

```diff
@@ -764,14 +777,14 @@ func (b *dbBufferBucket) merge() (mergeResult, error) {
 	}()
 
 	for i := range b.bootstrapped {
-		stream, err := b.bootstrapped[i].Stream(ctx)
-		if err == nil && stream != nil {
+		block, err := b.bootstrapped[i].Stream(ctx)
+		if err == nil && !block.IsEmpty() {
 			merges++
-			readers = append(readers, stream)
+			readers = append(readers, block)
```

Comment:
Maybe `append(readers, block.SegmentReader())` to avoid alloc of the block reader struct here too?